From 96135e3fabc820d37c9b10c78f6b2dbc22c6a9a5 Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.de>
Date: Thu, 18 Jun 2015 16:59:40 +0200
Subject: [PATCH] qat: fix different cpuinfo for Xen
Patch-mainline: Never (our Xen and upstream diverge, upstream doesn't need this)
References: FATE#318531

Xen doesn't provide phys_proc_id. Get around by assuming 0.

Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/crypto/qat/qat_common/qat_algs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/crypto/qat/qat_common/qat_algs.c b/drivers/crypto/qat/qat_common/qat_algs.c
index 542b0f7..53d2630 100644
--- a/drivers/crypto/qat/qat_common/qat_algs.c
+++ b/drivers/crypto/qat/qat_common/qat_algs.c
@@ -137,7 +137,11 @@ struct qat_alg_ablkcipher_ctx {
 
 static int get_current_node(void)
 {
+#ifdef CONFIG_XEN
+	return 0;
+#else
 	return cpu_data(current_thread_info()->cpu).phys_proc_id;
+#endif
 }
 
 static int qat_get_inter_state_size(enum icp_qat_hw_auth_algo qat_hash_alg)
-- 
2.1.4

